pkg/cvo/internal/generic_test.go

```diff
@@ -87,3 +87,71 @@ func TestCreateOnlyUpdate(t *testing.T) {
 		t.Error("should not have updated")
 	}
 }
+
+func TestCreateNoMatchError(t *testing.T) {
+	for _, tc := range []struct {
+		name     string
+		resource string
+	}{
+		{
+			name: "basic",
+			resource: `{
+  "kind": "UnknownType",
+  "apiVersion": "config.openshift.io/v1",
+  "metadata": {
+     "name": "cluster"
+  }
+}`,
+		},
+		{
+			name: "delete annotation",
+			resource: `{
+  "kind": "UnknownType",
+  "apiVersion": "config.openshift.io/v1",
+  "metadata": {
+     "name": "cluster",
+	 "annotations": {
```

Comment:
nit: seems like you're adding an additional level of indentation here?

Reply:
Mixing tabs and spaces got me there. Fixed now.